core/src/main/java/hudson/security/csrf/CrumbFilter.java

```diff
@@ -68,18 +68,22 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
                 // compatibility for clients that hard-code the default crumb name up to Jenkins 1.TODO
                 extractCrumbFromRequest(httpRequest, ".crumb");
             }
+
+            // JENKINS-40344: Don't spam the log just because a session is expired
+            Level level = Jenkins.getAuthentication() == Jenkins.ANONYMOUS ? Level.FINE : Level.WARNING;
+
             if (crumb != null) {
                 if (crumbIssuer.validateCrumb(httpRequest, crumbSalt, crumb)) {
                     valid = true;
                 } else {
-                    LOGGER.log(Level.WARNING, "Found invalid crumb {0}.  Will check remaining parameters for a valid one...", crumb);
+                    LOGGER.log(level, "Found invalid crumb {0}.  Will check remaining parameters for a valid one...", crumb);
```

Comment:
Maybe it makes sense to also print authentication in the log, so the admin will know the user

Reply:
Only one message has been patched, but it's fine